load-testing/tests/relays_stress_test.go

```diff
@@ -0,0 +1,457 @@
+	s.ensureStakedActors(txResults, MsgStakeGateway, gateways)
+	s.ensureStakedActors(txResults, MsgStakeApplication, applications)
+
+	logger.Info().Msg("Actors staked")
+
+	// Update the list of staked suppliers.
+	s.stakedSuppliers = append(s.stakedSuppliers, suppliers...)
+
+	// Delegate the initial applications to the initial gateways
+	s.sendDelegateInitialAppsTxs(applications, gateways)
+	txResults = s.waitForTxsToBeCommitted()
+	s.ensureDelegatedApps(txResults, applications, gateways)
+
+	logger.Info().Msg("Apps delegated")
+
+	// Applications and gateways are now ready and will be active in the next session.
+	s.preparedApplications = append(s.preparedApplications, applications...)
+	s.preparedGateways = append(s.preparedGateways, gateways...)
+
+	// batchInfoObs maps session information to batch information used to schedule
+	// the relay requests to be sent on the current block.
+	batchInfoObs, batchInfoPublishCh := channel.NewReplayObservable[*batchInfoNotif](s.ctx, 5)
+	s.batchInfoObs = batchInfoObs
+
+	// sessionInfoObs asynchronously maps committed blocks to a notification which
+	// includes the session number and the start and end block heights of the session.
+	// It runs at the same frequency as committed blocks (i.e. 1:1).
+	s.sessionInfoObs = channel.Map(
+		s.ctx,
+		s.blockClient.CommittedBlocksSequence(s.ctx),
+		s.mapSessionInfoFn(batchInfoPublishCh),
+	)
+
+	// stakingObs asynchronously maps session information to a set of newly staked
+	// actor accounts, only notifying when new actors were staked and skipping otherwise.
+	// It stakes new suppliers & gateways but only funds new applications as they can't be
+	// delegated until after the respective gateway stake txs have been committed.
+	// It receives at the same frequency as committed blocks (i.e. 1:1) but only sends
+	// conditionally as described here.
+	stakingObs := channel.Map(s.ctx, s.sessionInfoObs, s.mapStakingInfoFn(plan))
+
+	// stakedAndDelegatingObs asynchronously maps over the staking info, notified
+	// when one or more actors have been newly staked. For each notification received,
+	// it waits for the new actors' staking/funding txs to be committed before sending
+	// staking & delegation txs for new applications.
+	stakedAndDelegatingObs := channel.Map(s.ctx, stakingObs,
+		func(ctx context.Context, notif *stakingInfoNotif) (*stakingInfoNotif, bool) {
+			// Ensure that new gateways and suppliers are staked.
+			// Ensure that new applications are funded and have an account entry on-chain
+			// so that they can stake and delegate in the next block.
+			txResults = s.waitForTxsToBeCommitted()
+			s.ensureFundedActors(txResults, notif.newApps)
+			s.ensureStakedActors(txResults, MsgStakeGateway, notif.newGateways)
+			s.ensureStakedActors(txResults, MsgStakeSupplier, notif.newSuppliers)
+
+			// Update the list of staked suppliers.
+			s.stakedSuppliers = append(s.stakedSuppliers, notif.newSuppliers...)
```

Comment:
NIT: Why do we use `staked` as the prefix for suppliers but `prepared` as the prefix for `apps` and `gateways`.

I realize that it's because there are extra steps for them to be "ready for testing", but think that should be hidden from the global naming.